src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ReplaceMethodWithProperty;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.ReplaceMethodWithProperty;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.ReplaceMethodWithProperty
+{
+    public class ReplaceMethodWithPropertyTests : AbstractCSharpCodeActionTest
+    {
+        protected override object CreateCodeRefactoringProvider(Workspace workspace)
+        {
+            return new ReplaceMethodWithPropertyCodeRefactoringProvider();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithGetName()
+        {
+            Test(
+@"class C { int [||]GetFoo() { } }",
+@"class C { int Foo { get { } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithoutGetName()
+        {
+            Test(
+@"class C { int [||]Foo() { } }",
+@"class C { int Foo { get { } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithArrowBody()
+        {
+            Test(
+@"class C { int [||]GetFoo() => 0; }",
+@"class C { int Foo { get; } => 0; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithoutBody()
+        {
+            Test(
+@"class C { int [||]GetFoo(); }",
+@"class C { int Foo { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithModifiers()
+        {
+            Test(
+@"class C { public static int [||]GetFoo() { } }",
+@"class C { public static int Foo { get { } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestMethodWithAttributes()
+        {
+            Test(
+@"class C { [A]int [||]GetFoo() { } }",
+@"class C { [A]int Foo { get { } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestExplicitInterfaceMethod()
+        {
+            Test(
+@"class C { int [||]I.GetFoo() { } }",
+@"class C { int I.Foo { get { } } }");
```

Comment:
> I think we should probably go the extra mile of changing the interface as well as cascading through to callsites (a la rename). 

That's what the implementation does.  I'll add an additional test ot make that clear :)